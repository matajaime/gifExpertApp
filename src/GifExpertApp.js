import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { ImageGrid } from "./components/ImageGrid";

const GifExpertApp = () => {
    
    const [categories, setCategories] = useState([]);

    const handleAdd = (newCategory) => {
        for ( let i = 0; i < categories.length; ++i ) {
            if ( categories[i] === newCategory ) {
                alert('Ya estÃ¡ agregado');
                return;
            }
        }
        setCategories( [newCategory, ...categories]);
    };

    return (
        <>
        <h2>Gif Expert App</h2>
        <AddCategory addCategory={handleAdd}/>
        <hr/>
            { categories.map( category => (
                <ImageGrid key={category} category={category} />
            ))}
        </>
    );
};

export default GifExpertApp;
