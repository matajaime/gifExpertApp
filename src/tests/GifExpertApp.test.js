import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('GifExpertApp tests',()=>{
    test('GifExpertApp snap shot init state',()=>{
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('GifExpertApp debe generar lista de ImageGrids',()=>{
        const categories = ['Mario','Zelda'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        const imageGrids = wrapper.find('ImageGrid');
        expect( imageGrids.length ).toBe( categories.length );
    });
});