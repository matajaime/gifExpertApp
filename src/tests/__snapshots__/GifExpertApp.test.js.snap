// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests GifExpertApp snap shot init state 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
</Fragment>
`;
